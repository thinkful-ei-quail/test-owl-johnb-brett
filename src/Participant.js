import React from 'react'

function Participant(props) {

    const style = {
        fontSize: 50,
        color: null,
    }

    if(props.inSession) {
        style.color = 'green'
    } else {
        style.color = 'lightgrey'
    }

    return (
        <div>
            <img src={props.avatar} alt='avatar'/>
            <h3>{props.name}</h3>
            <span style={style}>&#8226;</span>
            <span>{props.onStage ? 'on stage' : 'observing'}</span>
        </div>
    )
}

export default Participant