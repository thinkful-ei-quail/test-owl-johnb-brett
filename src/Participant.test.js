import React from 'react'
import ReactDOM from 'react-dom'
import Participant from './Participant'
import renderer from 'react-test-renderer'

it('should render without crashing', () => {
   const div = document.createElement('div')
   ReactDOM.render(<Participant />, div)
   ReactDOM.unmountComponentAtNode(div) 
})

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Participant name='john' inSession onStage />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})