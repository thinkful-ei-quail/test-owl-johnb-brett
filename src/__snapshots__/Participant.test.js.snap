// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <img
    alt="avatar"
  />
  <h3>
    john
  </h3>
  <span
    style={
      Object {
        "color": "green",
        "fontSize": 50,
      }
    }
  >
    â€¢
  </span>
  <span>
    on stage
  </span>
</div>
`;
